rxfeedback/src/main/java/org/notests/rxfeedback/AsyncSynchronized.kt

```diff
@@ -0,0 +1,36 @@
+package org.notests.rxfeedback
+
+private typealias Mutate<State> = (State) -> Unit
+
+internal class AsyncSynchronized<State>(private val state: State) {
+
+    private val queue = mutableListOf<Mutate<State>>()
+
+    fun async(mutate: (State) -> Unit) {
```

Comment:
Should this be named `async`? :)

Reply:
@markotron ,
mmm... Do you have any other suggestion? It's called a `async` because it the mutation to the state can be performed after the `async` completes.

Reply:
Yeah, that’s what confused me. It *can* be performed after the `async` completes. But in some cases it want. Actually it will last even longer (unbounded) if other threads keep queuing tasks. When you say that something is async, I would assume that it will just queue something and complete. 

So I would name this something like `queueOrExecuteAll` (or something similar).

Reply:
>Actually it will last even longer (unbounded) if other threads keep queuing tasks. When you say that something is async, I would assume that it will just queue something and complete.

Good point! I will rename it to `queueOrExecuteAll`.  It communicates the intent better.